frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
+            </div>
+        </th>
+    );
+};
+
+const MessageReceiversRow = ({
+    receiver,
+    activeColumn,
+    activeColumnSortOrder,
+    setModalText,
+    modalRef,
+}: MessageReceiversRowProps) => {
+    // We highlight the entire active column so within the row,
+    // we need to know which column is active
+    const checkForActiveColumn = (colName: string) =>
+        colName === activeColumn &&
+        activeColumnSortOrder !== FilterOptionsEnum.None
+            ? "active-col-td"
+            : "";
+    const uppercaseMain = receiver.Main.toUpperCase();
+    return (
+        <tr>
+            <td className={checkForActiveColumn(ColumnDataEnum.Name)}>
+                {receiver.Name}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Service)}>
+                {receiver.Service}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Date)}>
+                {receiver.Date}
+            </td>
+            <td
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.ReportId
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.ReportId}:`,
+                        body: receiver.ReportId,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.ReportId}
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Main)}>
+                <p
+                    className={`font-mono-sm border-1px bg-primary-lighter radius-md padding-left-1 padding-right-1 margin-top-0 ${
+                        uppercaseMain === StatusEnum.BATCH && "bg-blue-5"
+                    } ${uppercaseMain === StatusEnum.PROCESS && "bg-blue-10"} ${
+                        uppercaseMain === StatusEnum.READY && "bg-blue-20"
+                    }`}
+                >
+                    {uppercaseMain}
+                </p>
+            </td>
+            <td className={checkForActiveColumn(ColumnDataEnum.Sub)}>
+                {receiver.Sub}
+            </td>
+            <td
+                className={`message-receiver-break-word white-space-pre ${checkForActiveColumn(
+                    ColumnDataEnum.FileName
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.FileName}:`,
+                        body: receiver.FileName,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.FileName}
+            </td>
+            <td
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.TransportResults
+                )}`}
+                onClick={() => {
```

Comment:
Minor point, but `td` isn't an interactive element for accessibility.  Any chance this could be turned into an unstyled `<button>` instead?

Reply:
I think that would affect the table layout, I was thinking I could just add `role="button"`?

Reply:
Right sorry, I meant that the cell content could be wrapped in a button since it looks like it's all just phrasing content 🤔